lib/safe_cstruct.ml

```diff
@@ -0,0 +1,45 @@
+let to_string ?(off = 0) t =
+    try
+        Cstruct.to_string ~off t
+    with
+    | Invalid_argument _ -> ""
```

Comment:
I'm not sure hiding errors this way is a good idea. Doesn't this code mean we could end up with a PGP object from a file which isn't really a PGP key?  Debugging would also be harder if no exception is raised in case of a parser bug.